### PR TITLE
feat: Member.display_banner

### PR DESCRIPTION
**Pull request by MattyTheHacker**
## Summary

Implement Member.display_banner as per #2516 

## Information

- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed).
- [ ] This PR is **not** a code change (e.g. documentation, README, typehinting,
      examples, ...).

## Checklist

- [x] I have searched the open pull requests for duplicates.
- [x] If code changes were made then they have been tested.
  - [ ] I have updated the documentation to reflect the changes.
- [x] If `type: ignore` comments were used, a comment is also left explaining why.
- [x] I have updated the changelog to include these changes.

### CODE REVIEW

**Review by Lulalaby — CHANGES_REQUESTED** [review]
Changelog

File: CHANGELOG.md [hunk 1]
```diff
@@ -22,6 +22,9 @@ These changes are available on the `master` branch, but have not yet been releas
 
 ### Added
 
+- Added `Member.display_banner` method.
+  ([#2556](https://github.com/Pycord-Development/pycord/pull/2556))
+
```
Comment (Lulalaby) [inline CHANGELOG.md:27]:
```suggestion
```
We don't do newlines between entries

**Review by Lulalaby — CHANGES_REQUESTED** [review]

**Review by Lulalaby — APPROVED** [review]

File: discord/asset.py [hunk 2]
```diff
@@ -212,6 +212,19 @@ def _from_guild_avatar(
...
 
+    @classmethod
+    def _from_guild_banner(
+        cls, state, guild_id: int, member_id: int, banner: str
+    ) -> Asset:
+        animated: bool = banner.startswith("a_")
+        format: str = "gif" if animated else "png"
+        return cls(
+            state,
+            url=f"{cls.BASE}/guilds/{guild_id}/users/{member_id}/banners/{banner}.{format}?size=1024",
```
[the remark this answers is not in the document]
Comment (MattyTheHacker) [reply discord/asset.py:223]: Do you have a preference as to which to use? I did `1024` because that's what `_from_guild_avatar` used.

**Comment by Dorukyum** [comment]
I changed both sizes to 512 for consistency.

**Review by Dorukyum — CHANGES_REQUESTED** [review]
Testing required

**Comment by yoggys** [comment]
> Testing required

its missing some things, rn not working

File: discord/member.py [hunk 3]
```diff
@@ -603,6 +603,31 @@ def guild_avatar(self) -> Asset | None:
...
+        """Returns an :class:`Asset` for the guild banner
+        the member has. If unavailable, ``None`` is returned.
+
+        .. versionadded:: 2.7
+        """
+        if self._user.banner is None:
+            return None
+        return Asset._from_guild_banner(
+            self._state, self.guild.id, self.id, self._user.banner
+        )
```
Comment (yoggys) [inline discord/member.py:629]:
```suggestion
        if self._banner is None:
            return None
        return Asset._from_guild_banner(
            self._state, self.guild.id, self.id, self._banner
        )
```

You should also add:
- to __init__
`self._banner: str | None = data.get("banner")`
- to _copy
`self._banner = member._banner`
- to _update
`self._banner = data.get("banner")`

**Review by yoggys — CHANGES_REQUESTED** [review]

**Comment by yoggys** [comment]
@MattyTheHacker please also include:
```py
__slots__ = (
        ...
        "_banner",
        ...
    )
```

**Comment by yoggys** [comment]
working good rn 👍

**Review by yoggys — APPROVED** [review]

**Comment by NeloBlivion** [comment]
Thank you! Some notes:
- Main PR is still a draft, so might need to wait before merging https://github.com/discord/discord-api-docs/pull/6986
- When adding new fields from the API, make sure the respective TypedDict in `types` is updated (in this case, `types/member.py` -> `Member`)

**Comment by Lulalaby** [comment]
We don't need to wait for the pr because it's live in stable already. Discord is just behind schedule

**Review by plun1331 — DISMISSED** [review]

**Review by Lulalaby — APPROVED** [review]

**Review by Dorukyum — APPROVED** [review]

**Pull request merged**